atoma-service/src/handlers/mod.rs

```diff
@@ -572,19 +572,42 @@ pub mod vllm_metrics {
     static METRICS_CACHE: LazyLock<MetricsCache> = LazyLock::new(|| MetricsCache::new());
 
     /// Start the background task to update metrics every 30 seconds
+    ///
+    /// # Arguments
+    ///
+    /// * `chat_completions_service_urls` - A vector of tuples containing the chat completions service URL and the job name.
+    /// * `metrics_update_interval` - The interval in seconds to update the metrics.
+    #[instrument(level = "info", skip_all)]
     pub fn start_metrics_updater(
         chat_completions_service_urls: Vec<(String, String)>,
         metrics_update_interval: Option<u64>,
     ) {
-        let chat_completions_service_urls = Arc::new(chat_completions_service_urls);
+        type ChatCompletionsServiceUrls = Vec<(String, String)>;
+        let (vllm_chat_completions_service_urls, sglang_chat_completions_service_urls): (
+            ChatCompletionsServiceUrls,
+            ChatCompletionsServiceUrls,
+        ) = chat_completions_service_urls
+            .iter()
+            .cloned()
+            .partition(|(_, job)| job.contains("vllm"));
+        let vllm_chat_completions_service_urls = Arc::new(vllm_chat_completions_service_urls);
+        let sglang_chat_completions_service_urls = Arc::new(sglang_chat_completions_service_urls);
         tokio::spawn(async move {
             let metrics_interval = metrics_update_interval.unwrap_or(30);
             let mut interval = time::interval(Duration::from_secs(metrics_interval));
             loop {
                 interval.tick().await;
-                let metrics = get_metrics(&HTTP_CLIENT, &chat_completions_service_urls).await;
-                if metrics.iter().any(std::result::Result::is_ok) {
-                    METRICS_CACHE.update_metrics(metrics).await;
+                let vllm_metrics =
+                    get_metrics_vllm(&HTTP_CLIENT, &vllm_chat_completions_service_urls).await;
+                let sglang_metrics =
+                    get_metrics_sglang(&HTTP_CLIENT, &sglang_chat_completions_service_urls).await;
+                if vllm_metrics.iter().any(std::result::Result::is_ok) {
+                    METRICS_CACHE.update_metrics(vllm_metrics).await;
+                } else {
+                    tracing::warn!("Failed to retrieve any valid metrics, not updating cache");
```

Comment:
Let's make this warning specific to `vllm` and likewise the latter to `sglang`

```suggestion
            tracing::warn!("Failed to retrieve any valid VLLM metrics, not updating cache");
```